src/scripts/main.js

```diff
@@ -1,3 +1,79 @@
 'use strict';
 
-// write code here
+const addRowBtn = document.querySelector('.append-row');
+const remRowBtn = document.querySelector('.remove-row');
+const addColBtn = document.querySelector('.append-column');
+const remColBtn = document.querySelector('.remove-column');
+const table = document.querySelector('table');
+const tBody = table.querySelector('tbody');
+
+function updateButtonStates() {
+  if (tBody.childElementCount >= 10) {
+    addRowBtn.setAttribute('disabled', 'disabled');
+  } else {
+    addRowBtn.removeAttribute('disabled');
+  }
+
+  if (tBody.childElementCount <= 2) {
+    remRowBtn.setAttribute('disabled', 'disabled');
+  } else {
+    remRowBtn.removeAttribute('disabled');
+  }
+
+  if (tBody.firstElementChild.childElementCount >= 10) {
+    addColBtn.setAttribute('disabled', 'disabled');
+  } else {
+    addColBtn.removeAttribute('disabled');
+  }
+
+  if (tBody.firstElementChild.childElementCount <= 2) {
```

Comment:
Ensure that the table has at least one row and one column initially. If the table starts with fewer than two columns, attempting to remove a column could lead to errors. Consider adding a check to prevent removing columns if there are fewer than two.